base/missing.jl

```diff
@@ -188,3 +187,30 @@ function Base.iterate(itr::SkipMissing, state...)
     end
     item, state
 end
+
+"""
+    coalesce(x, y...)
+
+Return the first value in the arguments which is not equal to [`missing`](@ref),
+if any. Otherwise return `missing`.
+
+# Examples
+
+```jldoctest
+julia> coalesce(missing, 1)
+1
+
+julia> coalesce(1, missing)
+1
+
+julia> coalesce(nothing, 1)  # returns `nothing`
+
+julia> coalesce(missing, missing)
+missing
+```
+"""
+function coalesce end
+
+coalesce() = missing
```

Comment:
I'd find it more natural to define `coalesce(x::Missing) = throw(...)`. There's no reason to define a function with zero arguments AFAICT. Same remark for `something`.

Reply:
This isn't ancient Rome; we understand zero now :grin:

Reply:
I 100% agree. Zero things is missing the same way the sum of no things is zero and the product of no things is one. If we knew the type in those cases, we would absolutely define the empty sums and products to do that; in this case we do know the type and can return `missing`.

Reply:
OK, why not, though I'm not sure in what cases it could be useful. OTOH I don't think it makes sense for `something` (since an error is thrown), but let's discuss it in the other thread above.